content/en/docs/components/pipelines/sdk/v2/build-pipeline.md

```diff
@@ -0,0 +1,522 @@
+    pipeline parameters to experiment with different hyperparameters, such as
+    the learning rate used to train a model, or pass run-level inputs, such as
+    the path to an input file, into a pipeline run. The data type must be
+    specified for all pipeline parameters. 
+    
+    Use the factory functions created by
+    the `kfp.dsl.component` annotation and the
+    `kfp.components.load_component_from_url` function to create your pipeline's tasks. 
+    The inputs to the component factory functions can be pipeline parameters,
+    the outputs of other tasks, or a constant value. In this case, the
+    `web_downloader_task` task uses the `url` pipeline parameter, and the
+    `merge_csv_task` uses the `data` output of the `web_downloader_task`.
+
+    The `kfp.dsl.pipeline` annotation lets you specify the following:
+
+    *   `name`: The pipeline's name.
+    *   `description`: (Optional.) A description of the pipeline's workflow.  
+    *   `pipeline_root`: The default path where your pipeline's artifacts are
+         stored. This must be a path that your pipeline can read and write to,
+         such as a Persistent Volume Claim or a cloud service such as Google
+         Cloud Storage.
+
+
+```python
+# Define a pipeline and create a task from a component:
+@dsl.pipeline(
+    pipeline_root='gs://my-pipeline-root/example-pipeline',
+    name="example-pipeline",
+)
+def my_pipeline(url: str):
+  web_downloader_task = web_downloader_op(url=url)
+  merge_csv_task = merge_csv(tar_data=web_downloader_task.outputs['data'])
+  # The outputs of the merge_csv_task can be referenced using the
+  # merge_csv_task.outputs dictionary: merge_csv_task.outputs['output_csv']
+```
+
+### Compile and run your pipeline
+
+After defining the pipeline in Python as described in the preceding section, use the following instructions to compile the pipeline and submit it to the Kubeflow Pipelines service. 
+
+1.  Run the following to compile your pipeline and save it as `pipeline.yaml`. 
+
+
+
+```python
+compiler.Compiler().compile(
+    pipeline_func=my_pipeline,
+    package_path='pipeline.yaml')
+```
+
+2.  Run the following to submit the compiled workflow specification
+    (`pipeline.yaml`) using the Kubeflow Pipelines SDK. 
+    
+    You can also use the Kubeflow Pipelines user interface to upload and run
+    your `pipeline.yaml`. See the guide to [getting started with the
+    UI][quickstart].
+
+[quickstart]: https://www.kubeflow.org/docs/components/pipelines/pipelines-quickstart
+
+
+```python
+client.create_run_from_pipeline_package(
+    pipeline_file='pipeline.yaml',
+    arguments={
+        'url': 'https://storage.googleapis.com/ml-pipeline-playground/iris-csv-files.tar.gz'
+    },
+    mode=kfp.dsl.PipelineExecutionMode.V2_COMPATIBLE)
+```
+
+
+## Next steps
+
+*   Learn about advanced pipeline features, such as [authoring recursive
+    components][recursion] and [using conditional execution in a
+    pipeline][conditional].
```

Comment:
Remove `recursive components`? because it's not supported in v2 yet.
/cc @chensun

Reply:
Done